src/Selenium2Library/locators/windowmanager.py

```diff
@@ -53,21 +61,36 @@ def _select_by_url(self, browser, criteria):
 
     def _select_by_default(self, browser, criteria):
         if criteria is None or len(criteria) == 0 or criteria.lower() == "null":
-            browser.switch_to_window('')
+            handles = browser.get_window_handles()
+            browser.switch_to_window(handles[0])
             return
-
-        try:
-            self._select_by_name(browser, criteria)
-            return
-        except ValueError: pass
-
+        for handle in browser.get_window_handles():
+            browser.switch_to_window(handle)
```

Comment:
Wouldn't it be better to only switch to the window when you know that it meets the right criteria? If it's unable to locate the handle you want to switch to, I would think the expected behavior would be to stay on the current handle.